galasa-extensions-parent/dev.galasa.auth.couchdb/src/main/java/dev/galasa/auth/couchdb/internal/CouchdbAuthStoreValidator.java

```diff
@@ -6,26 +6,176 @@
+                        "Validation failed of database galasa_tokens design document - " + statusLine.toString());
+            }
+            if (statusLine.getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                docJson = "{}";
+            }
+        } catch (CouchdbException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new CouchdbException("Validation failed", e);
+        }
+
+        boolean updated = false;
+
+        JsonObject doc = gson.fromJson(docJson, JsonObject.class);
+        doc.remove("_id");
+        String rev = null;
+        if (doc.has("_rev")) {
+            rev = doc.get("_rev").getAsString();
+        }
+
+        JsonObject views = doc.getAsJsonObject("views");
+        if (views == null) {
+            updated = true;
+            views = new JsonObject();
+            doc.add("views", views);
+        }
+
+        JsonObject requestors = views.getAsJsonObject("loginId-view");
+        if (requestors == null) {
+            updated = true;
+            requestors = new JsonObject();
+            views.add("loginId-view", requestors);
+        }
+
+        if (checkView(requestors, "function (doc) { if (doc.owner && doc.owner.loginId) {emit(doc.owner.loginId, doc); } }", "javascript")) {
+            updated = true;
+        }
+
+        if (updated) {
+            logger.info("Updating the galasa_tokens design document");
+
+            HttpEntity entity = new StringEntity(gson.toJson(doc), ContentType.APPLICATION_JSON);
+
+            HttpPut httpPut = requestFactory.getHttpPutRequest(couchdbUri + "/galasa_tokens/_design/docs");
+            httpPut.setEntity(entity);
+
+            if (rev != null) {
+                httpPut.addHeader("ETaq", "\"" + rev + "\"");
+            }
+
+            try (CloseableHttpResponse response = httpClient.execute(httpPut)) {
+                StatusLine statusLine = response.getStatusLine();
+                int statusCode = statusLine.getStatusCode();
+                if (statusCode == HttpStatus.SC_CONFLICT) {
+                    // Someone possibly updated
+                    attempts++;
+                    if (attempts > 10) {
+                        throw new CouchdbException(
+                                "Update of galasa_token design document failed on CouchDB server due to conflicts, attempted 10 times");
+                    }
+                    Thread.sleep(1000 + new Random().nextInt(3000));
+                    checkTokensDesignDocument(httpClient, couchdbUri, attempts);
+                    return;
+                }
+                
+                if (statusCode != HttpStatus.SC_CREATED) {
+                    EntityUtils.consumeQuietly(response.getEntity());
+                    throw new CouchdbException(
+                            "Update of galasa_tokens design document failed on CouchDB server - " + statusLine.toString());
+                }
+
+                EntityUtils.consumeQuietly(response.getEntity());
```

Comment:
That seems duplicate code within the above if... and outside of it ? Just move it before the if... ?